umap/templates/umap/map_init.html

```diff
@@ -1,7 +1,9 @@
 {% load umap_tags %}
 <div id="map"></div>
 <!-- djlint:off -->
-<script type="text/javascript">
-  let MAP = new L.U.Map("map", {{ map_settings|notag|safe }})
+<script defer type="text/javascript">
+  window.addEventListener('load', (event) => {
```

Comment:
Had to add this, otherwise the objects wouldn't be exported to `window`.

Reply:
As far as I understand:
- `defer` will not work for local script (i.e. without a `src` attribute)
- module type scripts are executed in defer mode, which means after the DOM is ready

I guess this explain why you needed the `load` event here. Maybe we could use [`DomContentLoaded`](https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event) instead ?

One other option to consider (but not for this PR I'd say), is to have a dedicated script (`/map/xxxx/load.js`), which could then make it work in defer mode, and could be an elegant way for other people to include a uMap map in their HTML ?

Reply:
Interesting, `defer` is not useful here then. `DomContentLoaded` seems the way (if it works), and will allow us to run the script without having to wait for other resources to be ready (images, etc).